panopticum/signals.py

```diff
@@ -0,0 +1,21 @@
+from django.conf import settings
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from rest_framework.authtoken.models import Token
+from django.core.files.base import ContentFile
+import django_auth_ldap.backend
+
+
+@receiver(post_save, sender=settings.AUTH_USER_MODEL)
+def create_auth_token(sender, instance=None, created=False, **kwargs):
+    """ Create auth token at creation User """
+    if created:
+        Token.objects.create(user=instance)
+
+
+@receiver(django_auth_ldap.backend.populate_user, sender=django_auth_ldap.backend.LDAPBackend)
+def add_photo(sender, instance=None, created=False, user=None, ldap_user=None, **kwargs):
+    """ Download and assign user photo at user population from LDAP """
+    if ldap_user.attrs.data['thumbnailphoto'] and not user.photo:
+        user.photo.save(name=f'{user.username}.jpg',
```

Comment:
It is built-in behavior of ```imageField```. We can implement our ```imageFiled``` or use some ```blobField``` with raw data. With ```blobField``` we will lose presentation image at admin panel.
Also we can serve avatars like a static files by nginx. That allow to reduce load on django app and database.

let's postpone that question for the future.

Reply:
We'll need to implement a backup procedure then. Also it will be harder to clone an instance for temporary development, for example simple DB dump on linux / restore on mac will not work.